python/cudf/cudf/core/indexing.py

```diff
@@ -266,11 +266,17 @@ def _getitem_tuple_arg(self, arg):
             else:
                 return columns_df.index._get_row_major(columns_df, arg[0])
         else:
-            df = DataFrame()
-            for col in columns_df.columns:
-                # need Series() in case a scalar is returned
-                df[col] = Series(columns_df[col].loc[arg[0]])
-            df.columns = columns_df.columns
+            if hasattr(arg[0], "__len__"):
+                arg = (column.as_column(arg[0]), arg[1])
+
+            if pd.api.types.is_bool_dtype(arg[0]):
+                return columns_df.take(arg[0])
+            else:
+                df = DataFrame()
+                for col in columns_df.columns:
+                    # need Series() in case a scalar is returned
+                    df[col] = Series(columns_df[col].loc[arg[0]])
```

Comment:
This gathers the index every time from this as well and still does the gathers column by column unnecessarily. Could we just translate whatever this is to a gathermap and then feed it into `iloc`?

Reply:
Did you mean to use it for non-boolean argument as well ?

Reply:
I guess I mean that underneath the column `.loc` we're likely doing some join shenanigans to actually accomplish the operation, where we could just do that join against the dataframe as a whole to save all of the separate calls here, especially if we're building a hash table each time in the join.